File: java-example/src/main/java/org/onflow/examples/java/getNodeVersionInfo/GetNodeVersionInfoAccessAPIConnector.java
package org.onflow.examples.java.getNodeVersionInfo;

import org.onflow.flow.sdk.FlowAccessApi;
import org.onflow.flow.sdk.FlowNodeVersionInfo;

public class GetNodeVersionInfoAccessAPIConnector {
    private final FlowAccessApi accessAPI;

    public GetNodeVersionInfoAccessAPIConnector(FlowAccessApi accessAPI) {
        this.accessAPI = accessAPI;
    }

    public FlowNodeVersionInfo getNodeVersionInfo() {
        FlowAccessApi.AccessApiCallResponse<FlowNodeVersionInfo> response = accessAPI.getNodeVersionInfo();
        if (response instanceof FlowAccessApi.AccessApiCallResponse.Success) {
            return ((FlowAccessApi.AccessApiCallResponse.Success<FlowNodeVersionInfo>) response).getData();
        } else {
            FlowAccessApi.AccessApiCallResponse.Error errorResponse = (FlowAccessApi.AccessApiCallResponse.Error) response;
            throw new RuntimeException(errorResponse.getMessage(), errorResponse.getThrowable());
        }
    }
}

File: kotlin-example/src/main/kotlin/org/onflow/examples/kotlin/getNodeVersionInfo/GetNodeVersionInfoAccessAPIConnector.kt
package org.onflow.examples.kotlin.getNodeVersionInfo

import org.onflow.flow.sdk.FlowAccessApi
import org.onflow.flow.sdk.FlowNodeVersionInfo

internal class GetNodeVersionInfoAccessAPIConnector(
    private val accessAPI: FlowAccessApi
) {
    fun getNodeVersionInfo(): FlowNodeVersionInfo {
        return when (val response = accessAPI.getNodeVersionInfo()) {
            is FlowAccessApi.AccessApiCallResponse.Success -> response.data
            is FlowAccessApi.AccessApiCallResponse.Error -> throw Exception(response.message, response.throwable)
        }
    }
}
